src/ring_theory/ideal/local_ring.lean

```diff
@@ -371,6 +371,37 @@ lemma map_comp (f : T →+* R) (g : R →+* S) [is_local_ring_hom f] [is_local_r
   (local_ring.residue_field.map g).comp (local_ring.residue_field.map f) :=
 ideal.quotient.ring_hom_ext $ ring_hom.ext $ λx, rfl
 
+section
+
+open ring_equiv local_ring
+
+@[simp] lemma map_id_apply {x : residue_field R} : map (ring_hom.id R) x = x :=
+by simp only [map_id, ring_hom.id_apply]
+
+@[simp] lemma map_map (f : R →+* S) (g : S →+* T) (x : residue_field R)
+[is_local_ring_hom f] [is_local_ring_hom g] :
+  map g (map f x) = map (g.comp f) x :=
+by simp only [map_comp, ring_hom.comp_apply]
+
+noncomputable theory
+
+/-- A ring isomorphism defines an isomorphism of residue fields -/
+@[simps apply]
+def map_equiv (f : R ≃+* S):
+  local_ring.residue_field R ≃+* local_ring.residue_field S :=
+{ to_fun := map (f : R →+* S),
+  inv_fun := map (f.symm : S →+* R),
+  left_inv := λ x, by simp only [← map_map, symm_comp, map_id, ring_hom.id_apply],
```

Comment:
This error looks easy to fix to me; look carefully at the goal state and whether you want to apply `map_map` in the forward or backward direction

Reply:
Nice, thanks